types/block.go

```diff
@@ -108,10 +112,14 @@ func (b Block) ID() BlockID {
 func (b Block) MerkleRoot() crypto.Hash {
 	tree := crypto.NewTree()
 	for _, payout := range b.MinerPayouts {
-		tree.PushObject(payout)
+		var b bytes.Buffer
+		payout.MarshalSia(&b)
+		tree.Push(b.Bytes())
 	}
 	for _, txn := range b.Transactions {
-		tree.PushObject(txn)
+		var b bytes.Buffer
+		txn.MarshalSia(&b)
+		tree.Push(b.Bytes())
```

Comment:
I think you can do even better here. Create the buffer outside the loop and use `buf.Reset` to clear it. That way, it retains the underlying capacity, so it doesn't need to resize every time you write to it.